### ISSUE TITLE
Add licence verification

### ISSUE THREAD

**Issue by jbonnet** [sonata-nfv/tng-gtk-sp#199]
Check [this](https://git.cs.upb.de/5gtango/UserStories/issues/297) user story.

**Comment by mtouloup** [comment]
Summary of the process...

1) An instantiation request arrives to the GK
2) If there is a selected SLA in the request, then the GK should request the license status of the particular SLA - NS with this API request (`curl GET http://localhost:8080/tng-sla-mgmt/api/slas/v1/licenses/status/{sla_uuid}/{ns_uuid}`)
3) Response from the above api request: 
```json
{
    "current_instances": "0",
    "allowed_instances": "20",
    "allowed_to_instantiate": "false",
    "license_type": "private",
    "license_status": "test",
    "license_expiration_date": "2020-12-01T00:00:00Z"
}
```
4a)  If the field "allowed_to_instantiate" == TRUE you just promote the request for instantiation to the MANO

4b) If the field "allowed_to_instantiate" == FALSE

- If  "license_type": == "private" &&  "allowed_to_instantiate": "false" 

the GK must send another request in order to "buy" the license    
 (`curl -X POST -H "Content-type:application/x-www-form-urlencoded" -d "ns_uuid=<>&sla_uuid=<>&license_type=<>&license_exp_date=<>&license_period=<>&allowed_instances=<>" http://localhost:8080/tng-sla-mgmt/api/slas/v1/licenses/buy`)   ---> and then try to instatiate again    

- If  "license_type": == "public"  OR  "license_type": == "trial" AND "allowed_to_instantiate": "false" ---> the "problem" is the number of instances of the service --> meaning the current instances are >= allowed instances --> meaning that THIS specific customer CANNOT continue using THIS ns with THIS SLA. The instantiation is not allowed. So the GK should reply with a message like:: "Instantiation not Allowed. NS instances reached the maximum allowed number" 

I propose to first go with the simple approach (Steps 1 - 4a). And then we can go further!

**Comment by ekapassa** [comment]
Hello @jbonnet, do we have any update on this issue?? :)

**Comment by jbonnet** [comment]
ok, @ekapassa , started...

**Comment by jbonnet** [comment]
Closed via PR #362
